core/src/main/java/org/apache/calcite/rel/core/Correlate.java

```diff
@@ -90,13 +90,24 @@ protected Correlate(
       RelNode left,
       RelNode right,
       CorrelationId correlationId,
-      ImmutableBitSet requiredColumns, SemiJoinType joinType) {
+      ImmutableBitSet requiredColumns,
+      JoinRelType joinType) {
```

Comment:
Correlate must have as joinType one of INNER, LEFT, SEMI, ANTI (i.e the formerly SemiJoinTypes), I think we should probably throw an IllegalArgumentException if joinType is not one of those

Reply:
I agree, add join type check.